### PR TITLE
@mzikherman - Handles setting sailthru variables for artsy-primer

### PR DESCRIPTION
**Pull request by broskoski**
Still needs tests!

### CODE REVIEW

File: desktop/apps/artsy_primer/personalize/client/index.coffee [hunk 1]
```diff
@@ -76,10 +77,15 @@ module.exports.PersonalizeRouter = class PersonalizeRouter extends Backbone.Rout
 
     @$el.attr 'data-state', 'loading'
 
-    $.when.apply(null, [
-      @user.save()
-      $.post('/flash', message: 'Thank you. Please expect your personalized portfolio in the next 2 business days.')
-    ]).always =>
+    Q.all [
+      $.post('/artsy-primer/set-sailthru')
```
Comment (mzikherman) [inline desktop/apps/artsy_primer/personalize/client/index.coffee]: This doesn't need to be `Q.all` right?

File: desktop/apps/artsy_primer/personalize/client/index.coffee [hunk 2]
```diff
@@ -76,10 +77,15 @@ module.exports.PersonalizeRouter = class PersonalizeRouter extends Backbone.Rout
...
+    ]
+    .then =>
+      Q.all [
+        @user.save()
+        $.post('/flash', message: 'Thank you. Please expect your personalized portfolio in the next 2 business days.')
+      ]
+    .finally =>
```
Comment (mzikherman) [inline desktop/apps/artsy_primer/personalize/client/index.coffee]: nice

File: desktop/apps/artsy_primer/routes.js [hunk 3]
```diff
@@ -3,6 +3,9 @@ import _ from 'underscore'
 import Article from '../../models/article'
 import { crop } from '../../components/resizer'
 
+import { SAILTHRU_KEY, SAILTHRU_SECRET, SAILTHRU_MASTER_LIST } from '../../config.coffee'
```
Comment (mzikherman) [inline desktop/apps/artsy_primer/routes.js]: the word 'sailthru' is mentioned 8 times on these 2 lines

File: desktop/apps/artsy_primer/routes.js [hunk 4]
```diff
@@ -19,3 +22,22 @@ export const personalize = async (req, res) => {
   }
   res.render('personalize')
 }
+
+export const setSailthruData = async (req, res) => {
```
Comment (mzikherman) [inline desktop/apps/artsy_primer/routes.js]: Nice, so this should be enough to disable them from getting the emails we don't want them to? (ie- can be handled by email/ST team).

File: desktop/apps/artsy_primer/personalize/client/index.coffee [hunk 5]
```diff
@@ -76,10 +77,15 @@ module.exports.PersonalizeRouter = class PersonalizeRouter extends Backbone.Rout
...
-    $.when.apply(null, [
-      @user.save()
-      $.post('/flash', message: 'Thank you. Please expect your personalized portfolio in the next 2 business days.')
-    ]).always =>
+    Q.all [
+      $.post('/artsy-primer/set-sailthru')
+    ]
+    .then =>
+      Q.all [
+        @user.save()
```
Comment (mzikherman) [inline desktop/apps/artsy_primer/personalize/client/index.coffee]: if we want, we can also pass in `receive_personalized_email: false` thing too. I still don't understand _why_ we need to do that if we can just disable the emails via the `artsy_primer` var in ST, but at least their profile preferences will be correct?

File: desktop/apps/artsy_primer/routes.js (hunk 3 above)
Comment (broskoski) [inline desktop/apps/artsy_primer/routes.js]: Today I learned.

File: desktop/apps/artsy_primer/routes.js (hunk 4 above)
Comment (broskoski) [inline desktop/apps/artsy_primer/routes.js]: Right and https://github.com/artsy/force/pull/1073/files#diff-f6777e97912484892bb42cd8515b5023R113

File: desktop/apps/artsy_primer/personalize/client/index.coffee (hunk 5 above)
Comment (broskoski) [inline desktop/apps/artsy_primer/personalize/client/index.coffee]: Ah yes, see previous comment.

**Comment by mzikherman** [comment]
I guess we should test w/ Sara (in practice) that this _works_ and accomplishes what we want in the ST interactions, but it seems likely to me!

Nice and easy, lookin' awesome! 🎉

File: desktop/apps/artsy_primer/personalize/client/index.coffee (hunk 5 above)
Comment (mzikherman) [inline desktop/apps/artsy_primer/personalize/client/index.coffee]: got it 👁

**Comment by broskoski** [comment]
Think this is ready to go, closes https://github.com/artsy/force/pull/1073 (I hope).

**Comment by mzikherman** [comment]
Cool, lmk if you need help testing!

**Pull request merged**